provider/maas/environ.go

```diff
@@ -859,6 +861,24 @@ func (*maasEnviron) MaintainInstance(args environs.StartInstanceParams) error {
 	return nil
 }
 
+type statusCallbackFunc func(settableStatus status.Status, info string, data map[string]interface{}) error
+
+func pollinstanceStatus(done <-chan struct{}, statusCallback statusCallbackFunc, inst maasInstance) {
+	for {
+		instStatus := inst.Status()
+		err := statusCallback(instStatus.Status, instStatus.Message, nil)
+		if err != nil {
+			logger.Errorf("setting instance %s status: %v", inst.Id(), err)
+		}
+
+		select {
+		case <-done:
+			return
+		case <-time.After(statusPollInterval):
```

Comment:
Well, as I understand, @howbazaar states that I should either pass a clock all the way from whoever is instantiating environ or just use time, I believe that for a use of After, time is reasonable.